graylog2-server/src/main/java/org/graylog2/periodical/UserPermissionMigrationPeriodical.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.periodical;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.graylog2.plugin.database.ValidationException;
+import org.graylog2.plugin.database.users.User;
+import org.graylog2.plugin.periodical.Periodical;
+import org.graylog2.shared.security.RestPermissions;
+import org.graylog2.shared.users.UserService;
+import org.graylog2.users.RoleService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This task migrates users' permission sets from pre-1.2 style to the built-in roles: admin and reader.
+ *
+ * It does so by preserving the existing stream and dashboard permissions, which can still be assigned to individual users.
+ * It is recommend to migrate to using roles exclusively, but for now both styles will work.
+ */
+public class UserPermissionMigrationPeriodical extends Periodical {
+    private static final Logger log = LoggerFactory.getLogger(UserPermissionMigrationPeriodical.class);
+
+    private final UserService userService;
+    private final RoleService roleService;
+
+    @Inject
+    public UserPermissionMigrationPeriodical(final UserService userService, final RoleService roleService) {
+        this.userService = userService;
+        this.roleService = roleService;
+    }
+
+    @Override
+    public void doRun() {
+        final List<User> users = userService.loadAll();
+        final String adminRoleId = roleService.getAdminRoleObjectId();
+        final String readerRoleId = roleService.getReaderRoleObjectId();
+
+        for (User user : users) {
```

Comment:
will do, even though i think it will not make much of a difference, because it won't touch users after they are migrated.
still might be confusing people on debug level